FAILURE test_send_jsonld_request

```
=================================== FAILURES ===================================
___________________________ test_send_jsonld_request ___________________________

alice_member_client = <shared.util.rich_async_client.RichAsyncClient object at 0x7f1dc1baedc0>
faber_client = <shared.util.rich_async_client.RichAsyncClient object at 0x7f1dc1e4a910>
faber_and_alice_connection = FaberAliceConnect(alice_connection_id='f352e29a-aad8-4837-a7de-2385c8b7238a', faber_connection_id='5ba61e21-6192-40f8-a61f-9952f912fab1')
register_issuer_key_ed25519 = 'did:key:z6MkoC4gXvZU2j18kJT59ytzwRH72BfPt795mjkETcZnJ3eo'

    @pytest.mark.anyio
    async def test_send_jsonld_request(
        alice_member_client: RichAsyncClient,
        faber_client: RichAsyncClient,
        faber_and_alice_connection: FaberAliceConnect,
        register_issuer_key_ed25519: DidKey,
    ):
        alice_connection_id = faber_and_alice_connection.alice_connection_id
        faber_connection_id = faber_and_alice_connection.faber_connection_id
    
        # Updating JSON-LD credential did:key with proofType ed25519
        credential = deepcopy(credential_)
        credential["connection_id"] = faber_connection_id
        credential["ld_credential_detail"]["credential"][
            "issuer"
        ] = register_issuer_key_ed25519
    
        response = await faber_client.post(
            CREDENTIALS_BASE_PATH,
            json=credential,
        )
        credential_exchange = response.json()
        assert credential_exchange["protocol_version"] == "v2"
    
        assert await check_webhook_state(
            client=faber_client,
            topic="credentials",
            filter_map={
                "state": "offer-sent",
                "credential_id": credential_exchange["credential_id"],
            },
        )
    
        response = await alice_member_client.get(
            CREDENTIALS_BASE_PATH,
            params={"connection_id": alice_connection_id},
        )
    
>       credential_id = (response.json())[0]["credential_id"]
E       IndexError: list index out of range

app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py:234: IndexError
---------------------------- Captured stdout setup -----------------------------
2023-10-03 17:01:10.951 | DEBUG    | app.event_handling.sse_listener:__init__:23 | Starting SseListener for wallet id `2049b750-2b64-4a09-916f-96a29011db2a` and topic `connections` | 
2023-10-03 17:01:12.402 | DEBUG    | app.event_handling.sse_listener:__init__:23 | Starting SseListener for wallet id `d7e0fcd0-cf8f-4110-aca7-36824e79ca86` and topic `connections` | 
2023-10-03 17:01:12.644 | INFO     | app.services.trust_registry:register_actor:410 | Registering actor on trust registry | {'actor': {'id': 'test-actor-0.4610850658923954', 'name': 'Test Actor-0.4610850658923954', 'roles': ['issuer'], 'did': 'did:key:z6MkoC4gXvZU2j18kJT59ytzwRH72BfPt795mjkETcZnJ3eo', 'didcomm_invitation': None}}
2023-10-03 17:01:12.791 | INFO     | app.services.trust_registry:register_actor:438 | Successfully registered actor on trust registry. | {'actor': {'id': 'test-actor-0.4610850658923954', 'name': 'Test Actor-0.4610850658923954', 'roles': ['issuer'], 'did': 'did:key:z6MkoC4gXvZU2j18kJT59ytzwRH72BfPt795mjkETcZnJ3eo', 'didcomm_invitation': None}}
----------------------------- Captured stdout call -----------------------------
2023-10-03 17:01:13.185 | DEBUG    | app.event_handling.sse_listener:__init__:23 | Starting SseListener for wallet id `d7e0fcd0-cf8f-4110-aca7-36824e79ca86` and topic `credentials` | 
--------------------------- Captured stdout teardown ---------------------------
2023-10-03 17:01:13.444 | INFO     | app.services.trust_registry:remove_actor_by_id:451 | Removing actor from trust registry | {'actor_id': 'test-actor-0.4610850658923954'}
2023-10-03 17:01:13.573 | INFO     | app.services.trust_registry:remove_actor_by_id:477 | Successfully removed actor from trust registry. | {'actor_id': 'test-actor-0.4610850658923954'}
=============================== warnings summary ===============================
../usr/local/lib/python3.9/site-packages/uplink/_extras.py:4
  /usr/local/lib/python3.9/site-packages/uplink/_extras.py:4: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py::test_send_jsonld_request
============ 1 failed, 193 passed, 1 warning in 1891.99s (0:31:31) =============
```

--- a/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
+++ b/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
@@ -149,7 +149,17 @@ async def test_send_jsonld_bbs_oob(
     )
 
     oob_record = accept_response.json()
+    alice_connection_id = oob_record['connection_id']
 
+    assert await check_webhook_state(
+        client=alice_member_client,
+        topic="connections",
+        filter_map={
+            "state": "completed",
+            "connection_id": alice_connection_id,
+        },
+    )
+    
     assert_that(accept_response.status_code).is_equal_to(200)
     assert_that(oob_record).contains("created_at", "oob_id", "invitation")
     assert await check_webhook_state(

--- a/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
+++ b/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
@@ -291,18 +291,18 @@ async def test_issue_jsonld_bbs(
         },
     )
 
+    assert await check_webhook_state(
+        client=alice_member_client,
+        filter_map={"state": "offer-received"},
+        topic="credentials",
+    )
+
     response = await alice_member_client.get(
         CREDENTIALS_BASE_PATH,
         params={"connection_id": alice_connection_id},
     )
 
     credential_id = (response.json())[0]["credential_id"]
-
-    assert await check_webhook_state(
-        client=alice_member_client,
-        filter_map={"state": "offer-received"},
-        topic="credentials",
-    )
 
     request_response = await alice_member_client.post(
         f"{CREDENTIALS_BASE_PATH}/{credential_id}/request",

--- a/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
+++ b/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
@@ -194,6 +194,7 @@ async def test_send_jsonld_bbs_oob(
         topic="credentials",
         filter_map={
             "state": "offer-received",
+            "connection_id": alice_connection_id,
         },
     )
 

--- a/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
+++ b/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
@@ -5,6 +5,7 @@ from aries_cloudcontroller import Credential, LDProofVCDetail, LDProofVCDetailOp
 from assertpy import assert_that
 
 from app.models.issuer import SendCredential
+from app.routes.connections import router as con_router
 from app.routes.issuer import router as issuer_router
 from app.routes.oob import router as oob_router
 from app.tests.util.ecosystem_connections import FaberAliceConnect
@@ -14,6 +15,8 @@ from shared import RichAsyncClient
 
 CREDENTIALS_BASE_PATH = issuer_router.prefix
 OOB_BASE_PATH = oob_router.prefix
+CON = con_router.prefix
+
 
 credential_ = SendCredential(
     type="ld_proof",

--- a/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
+++ b/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
@@ -162,6 +162,16 @@ async def test_send_jsonld_bbs_oob(
     
     assert_that(accept_response.status_code).is_equal_to(200)
     assert_that(oob_record).contains("created_at", "oob_id", "invitation")
+
+    faber_con = await faber_client.get(
+        CON
+    )
+
+    faber_connections = faber_con.json()
+    for con in faber_connections:
+
+        if con['invitation_msg_id'] == invitation['@id']:
+            faber_connection_id = con['connection_id']
     assert await check_webhook_state(
         client=alice_member_client,
         topic="credentials",

--- a/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
+++ b/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
@@ -130,44 +130,15 @@ async def test_send_jsonld_bbs_oob(
     alice_member_client: RichAsyncClient,
     register_issuer_key_bbs: DidKey,
 ):
-    alice_connection_id = faber_and_alice_connection.alice_connection_id
-    faber_connection_id = faber_and_alice_connection.faber_connection_id
-
-    response = await alice_member_client.get(
-        CREDENTIALS_BASE_PATH,
-        params={"connection_id": alice_connection_id},
-    )
-    records = response.json()
-
-    # nothing currently in alice's records
-    assert len(records) == 0
-
-    # Updating JSON-LD credential did:key (bbs)
-    credential = deepcopy(credential_)
-    credential["connection_id"] = faber_connection_id
-    credential["ld_credential_detail"]["credential"]["issuer"] = register_issuer_key_bbs
-
-    # faber create offer
-    response = await faber_client.post(
-        CREDENTIALS_BASE_PATH + "/create-offer",
-        json=credential,
-    )
-
-    data = response.json()
-    assert_that(data).contains("credential_id")
-    assert_that(data).has_state("offer-sent")
-    assert_that(data).has_protocol_version("v2")
-
     invitation_response = await faber_client.post(
         OOB_BASE_PATH + "/create-invitation",
         json={
-            "create_connection": False,
+            "create_connection": True,
             "use_public_did": False,
-            "attachments": [
-                {"id": data["credential_id"][3:], "type": "credential-offer"}
-            ],
+            "attachments": [],
         },
     )
+
     assert_that(invitation_response.status_code).is_equal_to(200)
 
     invitation = (invitation_response.json())["invitation"]

--- a/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
+++ b/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
@@ -126,7 +126,6 @@ async def test_send_jsonld_key_bbs(
 @pytest.mark.anyio
 async def test_send_jsonld_bbs_oob(
     faber_client: RichAsyncClient,
-    faber_and_alice_connection: FaberAliceConnect,
     alice_member_client: RichAsyncClient,
     register_issuer_key_bbs: DidKey,
 ):

--- a/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
+++ b/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
@@ -177,6 +177,18 @@ async def test_send_jsonld_bbs_oob(
     credential = deepcopy(credential_)
     credential["connection_id"] = faber_connection_id
     credential["ld_credential_detail"]["credential"]["issuer"] = register_issuer_key_bbs
+
+    # Send credential
+    response = await faber_client.post(
+        CREDENTIALS_BASE_PATH,
+        json=credential,
+    )
+
+    data = response.json()
+    assert_that(data).contains("credential_id")
+    assert_that(data).has_state("offer-sent")
+    assert_that(data).has_protocol_version("v2")
+
     assert await check_webhook_state(
         client=alice_member_client,
         topic="credentials",

--- a/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
+++ b/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
@@ -148,7 +148,7 @@ async def test_send_jsonld_bbs_oob(
     )
 
     oob_record = accept_response.json()
-    alice_connection_id = oob_record['connection_id']
+    alice_connection_id = oob_record["connection_id"]
 
     assert await check_webhook_state(
         client=alice_member_client,
@@ -158,19 +158,16 @@ async def test_send_jsonld_bbs_oob(
             "connection_id": alice_connection_id,
         },
     )
-    
+
     assert_that(accept_response.status_code).is_equal_to(200)
     assert_that(oob_record).contains("created_at", "oob_id", "invitation")
 
-    faber_con = await faber_client.get(
-        CON
-    )
+    faber_con = await faber_client.get(CON)
 
     faber_connections = faber_con.json()
     for con in faber_connections:
-
-        if con['invitation_msg_id'] == invitation['@id']:
-            faber_connection_id = con['connection_id']
+        if con["invitation_msg_id"] == invitation["@id"]:
+            faber_connection_id = con["connection_id"]
 
     # Updating JSON-LD credential did:key (bbs)
     credential = deepcopy(credential_)

--- a/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
+++ b/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
@@ -230,18 +230,18 @@ async def test_send_jsonld_request(
         },
     )
 
+    assert await check_webhook_state(
+        client=alice_member_client,
+        filter_map={"state": "offer-received"},
+        topic="credentials",
+    )
+
     response = await alice_member_client.get(
         CREDENTIALS_BASE_PATH,
         params={"connection_id": alice_connection_id},
     )
 
     credential_id = (response.json())[0]["credential_id"]
-
-    assert await check_webhook_state(
-        client=alice_member_client,
-        filter_map={"state": "offer-received"},
-        topic="credentials",
-    )
 
     request_response = await alice_member_client.post(
         f"{CREDENTIALS_BASE_PATH}/{credential_id}/request",

--- a/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
+++ b/app/tests/e2e/issuer/did_key_bbs/test_ld_bbs.py
@@ -172,6 +172,11 @@ async def test_send_jsonld_bbs_oob(
 
         if con['invitation_msg_id'] == invitation['@id']:
             faber_connection_id = con['connection_id']
+
+    # Updating JSON-LD credential did:key (bbs)
+    credential = deepcopy(credential_)
+    credential["connection_id"] = faber_connection_id
+    credential["ld_credential_detail"]["credential"]["issuer"] = register_issuer_key_bbs
     assert await check_webhook_state(
         client=alice_member_client,
         topic="credentials",

--- a/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
+++ b/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
@@ -191,11 +191,13 @@ async def test_send_jsonld_oob(
     assert_that(data).contains("credential_id")
     assert_that(data).has_state("offer-sent")
     assert_that(data).has_protocol_version("v2")
+
     assert await check_webhook_state(
         client=alice_member_client,
         topic="credentials",
         filter_map={
             "state": "offer-received",
+            "connection_id": alice_connection_id,
         },
     )
 

--- a/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
+++ b/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
@@ -130,46 +130,16 @@ async def test_send_jsonld_oob(
     alice_member_client: RichAsyncClient,
     register_issuer_key_ed25519: DidKey,
 ):
-    alice_connection_id = faber_and_alice_connection.alice_connection_id
-    faber_connection_id = faber_and_alice_connection.faber_connection_id
-
-    response = await alice_member_client.get(
-        CREDENTIALS_BASE_PATH,
-        params={"connection_id": alice_connection_id},
-    )
-    records = response.json()
-
-    # nothing currently in alice's records
-    assert len(records) == 0
-
-    # Updating JSON-LD credential did:sov
-    credential = deepcopy(credential_)
-    credential["connection_id"] = faber_connection_id
-    credential["ld_credential_detail"]["credential"][
-        "issuer"
-    ] = register_issuer_key_ed25519
-
-    # faber create offer
-    response = await faber_client.post(
-        CREDENTIALS_BASE_PATH + "/create-offer",
-        json=credential,
-    )
-
-    data = response.json()
-    assert_that(data).contains("credential_id")
-    assert_that(data).has_state("offer-sent")
-    assert_that(data).has_protocol_version("v2")
 
     invitation_response = await faber_client.post(
         OOB_BASE_PATH + "/create-invitation",
         json={
-            "create_connection": False,
+            "create_connection": True,
             "use_public_did": False,
-            "attachments": [
-                {"id": data["credential_id"][3:], "type": "credential-offer"}
-            ],
+            "attachments": [],
         },
     )
+
     assert_that(invitation_response.status_code).is_equal_to(200)
 
     invitation = (invitation_response.json())["invitation"]

--- a/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
+++ b/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
@@ -163,6 +163,16 @@ async def test_send_jsonld_oob(
 
     assert_that(accept_response.status_code).is_equal_to(200)
     assert_that(oob_record).contains("created_at", "oob_id", "invitation")
+
+    faber_con = await faber_client.get(
+        CON
+    )
+
+    faber_connections = faber_con.json()
+    for con in faber_connections:
+
+        if con['invitation_msg_id'] == invitation['@id']:
+            faber_connection_id = con['connection_id']
     assert await check_webhook_state(
         client=alice_member_client,
         topic="credentials",

--- a/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
+++ b/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
@@ -226,18 +226,18 @@ async def test_send_jsonld_request(
         },
     )
 
+    assert await check_webhook_state(
+        client=alice_member_client,
+        filter_map={"state": "offer-received"},
+        topic="credentials",
+    )
+
     response = await alice_member_client.get(
         CREDENTIALS_BASE_PATH,
         params={"connection_id": alice_connection_id},
     )
 
     credential_id = (response.json())[0]["credential_id"]
-
-    assert await check_webhook_state(
-        client=alice_member_client,
-        filter_map={"state": "offer-received"},
-        topic="credentials",
-    )
 
     request_response = await alice_member_client.post(
         f"{CREDENTIALS_BASE_PATH}/{credential_id}/request",

--- a/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
+++ b/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
@@ -180,6 +180,17 @@ async def test_send_jsonld_oob(
     credential["ld_credential_detail"]["credential"][
         "issuer"
     ] = register_issuer_key_ed25519
+
+    # Send credential
+    response = await faber_client.post(
+        CREDENTIALS_BASE_PATH,
+        json=credential,
+    )
+
+    data = response.json()
+    assert_that(data).contains("credential_id")
+    assert_that(data).has_state("offer-sent")
+    assert_that(data).has_protocol_version("v2")
     assert await check_webhook_state(
         client=alice_member_client,
         topic="credentials",

--- a/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
+++ b/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
@@ -291,18 +291,18 @@ async def test_issue_jsonld_ed(
         },
     )
 
+    assert await check_webhook_state(
+        client=alice_member_client,
+        filter_map={"state": "offer-received"},
+        topic="credentials",
+    )
+
     response = await alice_member_client.get(
         CREDENTIALS_BASE_PATH,
         params={"connection_id": alice_connection_id},
     )
 
     credential_id = (response.json())[0]["credential_id"]
-
-    assert await check_webhook_state(
-        client=alice_member_client,
-        filter_map={"state": "offer-received"},
-        topic="credentials",
-    )
 
     request_response = await alice_member_client.post(
         f"{CREDENTIALS_BASE_PATH}/{credential_id}/request",

--- a/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
+++ b/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
@@ -127,7 +127,6 @@ async def test_send_jsonld_key_ed25519(
 @pytest.mark.anyio
 async def test_send_jsonld_oob(
     faber_client: RichAsyncClient,
-    faber_and_alice_connection: FaberAliceConnect,
     alice_member_client: RichAsyncClient,
     register_issuer_key_ed25519: DidKey,
 ):

--- a/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
+++ b/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
@@ -5,6 +5,7 @@ from aries_cloudcontroller import Credential, LDProofVCDetail, LDProofVCDetailOp
 from assertpy import assert_that
 
 from app.models.issuer import SendCredential
+from app.routes.connections import router as con_router
 from app.routes.issuer import router as issuer_router
 from app.routes.oob import router as oob_router
 from app.routes.wallet import router as wallet_router
@@ -16,6 +17,7 @@ from shared import RichAsyncClient
 CREDENTIALS_BASE_PATH = issuer_router.prefix
 OOB_BASE_PATH = oob_router.prefix
 WALLET = wallet_router.prefix
+CON = con_router.prefix
 
 credential_ = SendCredential(
     type="ld_proof",

--- a/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
+++ b/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
@@ -132,7 +132,6 @@ async def test_send_jsonld_oob(
     alice_member_client: RichAsyncClient,
     register_issuer_key_ed25519: DidKey,
 ):
-
     invitation_response = await faber_client.post(
         OOB_BASE_PATH + "/create-invitation",
         json={
@@ -152,7 +151,7 @@ async def test_send_jsonld_oob(
     )
 
     oob_record = accept_response.json()
-    alice_connection_id = oob_record['connection_id']
+    alice_connection_id = oob_record["connection_id"]
 
     assert await check_webhook_state(
         client=alice_member_client,
@@ -166,15 +165,12 @@ async def test_send_jsonld_oob(
     assert_that(accept_response.status_code).is_equal_to(200)
     assert_that(oob_record).contains("created_at", "oob_id", "invitation")
 
-    faber_con = await faber_client.get(
-        CON
-    )
+    faber_con = await faber_client.get(CON)
 
     faber_connections = faber_con.json()
     for con in faber_connections:
-
-        if con['invitation_msg_id'] == invitation['@id']:
-            faber_connection_id = con['connection_id']
+        if con["invitation_msg_id"] == invitation["@id"]:
+            faber_connection_id = con["connection_id"]
 
     # Updating JSON-LD credential did:key with proofType ed25519
     credential = deepcopy(credential_)

--- a/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
+++ b/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
@@ -150,6 +150,16 @@ async def test_send_jsonld_oob(
     )
 
     oob_record = accept_response.json()
+    alice_connection_id = oob_record['connection_id']
+
+    assert await check_webhook_state(
+        client=alice_member_client,
+        topic="connections",
+        filter_map={
+            "state": "completed",
+            "connection_id": alice_connection_id,
+        },
+    )
 
     assert_that(accept_response.status_code).is_equal_to(200)
     assert_that(oob_record).contains("created_at", "oob_id", "invitation")

--- a/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
+++ b/app/tests/e2e/issuer/did_key_ed/test_ld_ed25519.py
@@ -173,6 +173,13 @@ async def test_send_jsonld_oob(
 
         if con['invitation_msg_id'] == invitation['@id']:
             faber_connection_id = con['connection_id']
+
+    # Updating JSON-LD credential did:key with proofType ed25519
+    credential = deepcopy(credential_)
+    credential["connection_id"] = faber_connection_id
+    credential["ld_credential_detail"]["credential"][
+        "issuer"
+    ] = register_issuer_key_ed25519
     assert await check_webhook_state(
         client=alice_member_client,
         topic="credentials",

